bluzelled/src/main.cpp

```diff
@@ -63,33 +48,26 @@ unsigned long get_min_nodes()
     return s_min_nodes;
 }
 
-void http_service()
+int main(/*int argc,char *argv[]*/)
 {
-    const std::string address("127.0.0.1");
-    const std::string port("2999");
-    std::string const doc_root = "../web/";
-    http::server::server s(address, port, doc_root);
-    s.run();
-}
-
-int main(/*int argc,char *argv[]*/) {
     uint8_t numTasks = 2;
     get_mutex();
 
     boost::thread websocket_thread(WebSocketServer("127.0.0.1", 3000, 1));
-    boost::thread http_thread(http_service);
 
     std::stringstream ss;
     try
         {
+        http::server::ServiceManager http_manager("127.0.0.1","2999","../web/");
+        http_manager.run_async();
+
         s_nodes = create_nodes(numTasks, g_listener);
         wait_for_all_nodes_to_start(s_nodes);
-        make_introductions(s_nodes);
 
         while (!s_nodes.empty())
             {
             reaper(&s_nodes);
-            int num_of_new_nodes = number_of_nodes_to_create(s_min_nodes, s_max_nodes, s_nodes.size());
+            unsigned long num_of_new_nodes = number_of_nodes_to_create(s_min_nodes, s_max_nodes, s_nodes.size());
```

Comment:
Not  a big deal, codingstandard mentions all parameters must be on its own line.